File: app/src/main/java/com/sorcererxw/matthiasheiderichphotography/ui/others/Navigator.java
package com.sorcererxw.matthiasheiderichphotography.ui.others;

import com.sorcererxw.matthiasheiderichphotography.ui.activities.MainActivity;

/**
 * @description:
 * @author: Sorcerer
 * @date: 2016/12/16
 */

public class Navigator {
    // todo: finish navigator
    private Navigator(MainActivity mainActivity) {
        mMainActivity = mainActivity;
    }

    private static Navigator mNavigator;

    private MainActivity mMainActivity;

    public static void init(MainActivity mainActivity) {
        mNavigator = new Navigator(mainActivity);
    }

    public static Navigator getInstance() {
        return mNavigator;
    }

    public void toHomePage() {

    }

    public void toFavoritePage() {

    }

    public void toSettingsPage() {

    }

    public void toCollectionPage(String collectionName) {

    }
}

File: app/src/main/java/com/sorcererxw/matthiasheiderichphotography/ui/others/Navigator.kt
package com.sorcererxw.matthiasheiderichphotography.ui.others

import com.sorcererxw.matthiasheiderichphotography.ui.activities.MainActivity

/**
 * @description:
 * @author: Sorcerer
 * @date: 2016/12/16
 */

class Navigator// todo: finish navigator
private constructor(private val mMainActivity: MainActivity) {

    fun toHomePage() {

    }

    fun toFavoritePage() {

    }

    fun toSettingsPage() {

    }

    fun toCollectionPage(collectionName: String) {

    }

    companion object {

        var instance: Navigator? = null
            private set

        fun init(mainActivity: MainActivity) {
            instance = Navigator(mainActivity)
        }
    }
}